### PR TITLE
Merge changes from 2019.2.5 and 3000.3

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Changes from 2019.2.5 and 3000.3

### Merge requirements satisfied?

- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by Akm0d — DISMISSED** [review]

**Review by xeacott — DISMISSED** [review]

**Review by frogunder — DISMISSED** [review]

**Review by garethgreenaway — DISMISSED** [review]

**Review by s0undt3ch — APPROVED** [review]

File: tests/integration/master/test_clear_funcs.py [hunk 1]
```diff
@@ -150,8 +150,9 @@ def test_clearfuncs_config(self):
         }
         ret = clear_channel.send(msg, timeout=5)
         assert not os.path.exists(
-            self.evil_file_path
```
Comment (s0undt3ch) [inline tests/integration/master/test_clear_funcs.py:153]: This line should be as it is on the master branch.
Dunno why it only failed on windows ...

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/integration/master/test_clear_funcs.py (hunk 1 above)
Comment (dwoz) [reply tests/integration/master/test_clear_funcs.py:153]: Yeah, that fact that it didn't fail on windows is raising some concerns for me.

Comment (s0undt3ch) [reply tests/integration/master/test_clear_funcs.py:153]: It failed on windows but not on Linux.

Comment (s0undt3ch) [reply tests/integration/master/test_clear_funcs.py:153]: Oh, you edited :)

Comment (s0undt3ch) [reply tests/integration/master/test_clear_funcs.py:153]: Yes, concerning

**Pull request merged**